From da24a1ef7ff38fd5695bf13ec582eef0fd4dcb1b Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 15 May 2023 09:50:07 +0200
Subject: [PATCH 0156/1695] clk: stm32mp13: fix of CKPER clock disabling

The mask used to configure the CKPER MUX disabled was wrong and
unnecessary.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a0f0c4fecd008b6c9743e33a6695b9c48b0a9544 clk: stm32mp13: fix of CKPER clock disabling]
Change-Id: I4be84f413b5f5e99cfd44182a8555f5e603788c4
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk-stm32mp13.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp13.c b/core/drivers/clk/clk-stm32mp13.c
index 2812d0b4a..cd46808a8 100644
--- a/core/drivers/clk/clk-stm32mp13.c
+++ b/core/drivers/clk/clk-stm32mp13.c
@@ -1066,8 +1066,7 @@ static int stm32_clk_source_configure(struct clk_stm32_priv *priv)
 	 * => deactivate CKPER only after switching clock
 	 */
 	if (ckper_disabled) {
-		ret = stm32_clk_configure_mux(priv,
-					      CLK_CKPER_DISABLED & CMD_MASK);
+		ret = stm32_clk_configure_mux(priv, CLK_CKPER_DISABLED);
 		if (ret != 0)
 			return ret;
 	}
-- 
2.47.2

